Cabal/Distribution/Simple/Utils.hs

```diff
@@ -1411,11 +1412,24 @@ shortRelativePath from to =
 -- unchanged.
 dropExeExtension :: FilePath -> FilePath
 dropExeExtension filepath =
-  case splitExtension filepath of
-    (filepath', extension) | extension `elem` exeExtensions -> filepath'
-                           | otherwise                      -> filepath
-
--- | List of possible executable file extensions on the current platform.
+  -- System.FilePath's extension handling functions are horribly
+  -- inconsistent, consider:
+  --
+  --     isExtensionOf "" "foo"  == False but
+  --     isExtensionOf "" "foo." == True.
+  --
+  -- On the other hand stripExtension doesn't remove the empty extension:
+  --
+  --    stripExtension "" "foo." == Just "foo."
+  --
+  -- Since by "" in exeExtensions we mean 'no extension' anyways we can
+  -- just always ignore it here.
+  let exts = [ ext | ext <- exeExtensions, ext /= "" ] in
+  fromMaybe filepath $
```

Comment:
I'd wrote this as

```haskell
fromMaybe filepath $ do
  ext <- find (`isExtensionOf` filepath)
  stripExtension ext filepath
```

IMHO, pointfree is pointless here.

Reply:
Sure thing.